server/embed/config_test.go

```diff
@@ -507,3 +512,10 @@ func TestTLSVersionMinMax(t *testing.T) {
 		})
 	}
 }
+
+func TestUndefinedAutoCompactionModeValidate(t *testing.T) {
+	cfg := *NewConfig()
+	cfg.AutoCompactionMode = ""
+	err := cfg.Validate()
+	assert.NotEqual(t, nil, err)
```

Comment:
```suggestion
	require.Error(t, err)
```

Error asserts that a function returned an error (i.e. not `nil`).

Reply:
The current PR LGTM. Thanks!

I can reproduce it locally without `e.Close`.